src/components/matrixSize.jsx

```diff
@@ -17,10 +17,33 @@ export default class MatrixSize extends React.Component {
       this.setState({ numCols: event.target.value });
   }
   handleClick(){
-  firebase.database().ref('grids/' + this.state.gridId + '/numRows' ).set(parseInt(this.state.numRows));
-  firebase.database().ref('grids/' + this.state.gridId + '/numCols' ).set(parseInt(this.state.numCols));
+  firebase.database().ref('grids/' + this.state.gridId).once('value', gridSnapshot => {
```

Comment:
This is a lot... Can't you just wipe all of the data and reset. I feel if they are resizing they should expect for their work to be reset.